src/boot/efi/cpio.c

```diff
@@ -346,15 +345,15 @@ EFI_STATUS pack_cpio(
         if (!root)
                 return log_error_status_stall(EFI_LOAD_ERROR, L"Unable to open root directory.");
 
-        loaded_image_path = DevicePathToStr(loaded_image->FilePath);
-        if (!loaded_image_path)
-                return log_oom();
-
-        j = PoolPrint(L"%s" EXTRA_DIR_SUFFIX, loaded_image_path);
-        if (!j)
-                return log_oom();
+        if (!global_dropin_dir) {
```

Comment:
Please invert this if and no {} for single statement clauses.

Reply:
Or better yet, simplify by assigning `rel_dropin_dir` to `dropin_dir` (but keeping both vars) and keeping the `open_directory` call as is.

Reply:
Do you mean I should create another variable (not covered by `_cleanup_freepool_`) called `dropin_dir`? Or should I rename `global_dropin_dir` to `dropin_dir` & remove the `const`?

Reply:
Rename it and keep the const (cause it refers to the data staying untouched, not the pointer itself).

Reply:
Ah whoops you're right. Done